File: src/src/yjhwang/level_1/달리기경주.js
function solution(players, callings) {
    const rankingTable = new Map();
    const ranking = players.slice();
  
    players.forEach((player, index) => {
      rankingTable.set(player, { player, index });
    });
  
    for (let i = 0; i < callings.length; i++) {
      const callingPlayer = rankingTable.get(callings[i]);
      const frontPlayer = rankingTable.get(ranking[callingPlayer.index - 1]);
  
      ranking[callingPlayer.index - 1] = callingPlayer.player;
      ranking[callingPlayer.index] = frontPlayer.player;
      rankingTable.set(callingPlayer.player, {
        player: callingPlayer.player,
        index: callingPlayer.index - 1,
      });
      rankingTable.set(frontPlayer.player, {
        player: frontPlayer.player,
        index: frontPlayer.index + 1,
      });
    }
  
    console.log(ranking);
  
    return ranking;
  }

File: src/src/bmkim/level_1/달리기경주.java
public class 달리기경주 {
    public static String[] solution(String[] players, String[] callings) {
        for (String calling : callings) {
            for (int i = 0; i < players.length; i++) {
                if (players[i].equals(calling)) {
                    if (i > 0) {
                        String temp = players[i];
                        players[i] = players[i - 1];
                        players[i - 1] = temp;
                    }
                    break;
                }
            }
        }
        return players;
    }
}
